ast/src/gen/generic.rs

```diff
@@ -2379,3 +2380,44 @@ impl<R> Node for TypeIgnore<R> {
     const NAME: &'static str = "type_ignore";
     const FIELD_NAMES: &'static [&'static str] = &[];
 }
+
+/// An alternative type of AST `arguments`. This is parser-friendly definition of function arguments.
+/// `defaults` and `kw_defaults` are placed under each `function_arg` typed fields.
+///
+/// NOTE: This type is different from original Python AST.
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct FunctionArguments<R = TextRange> {
+    pub range: OptionalRange<R>,
+    pub posonlyargs: Vec<FunctionArg<R>>,
+    pub args: Vec<FunctionArg<R>>,
+    pub vararg: Option<Box<Arg<R>>>,
+    pub kwonlyargs: Vec<FunctionArg<R>>,
+    pub kwarg: Option<Box<Arg<R>>>,
+}
+
+impl<R> Node for FunctionArguments<R> {
```

Comment:
What's the motivation of renaming the nodes rather than using Arguments and Argument?

Reply:
There were multiple reasons to keep the original types at least for a while.

About Arg, Both `Arg` and `FunctionArg` is used by FunctionArguments. because vararg and kwarg doens't have Default. Only `Arguments` is redundant type.
Renaming it makes the name goes different with original Python AST. It will prevent confusion.

The expected AST is the original node. The conversion must be easy. Keeping original types makes it easier to provide conversions. It also helps easy migration start just by calling conversion function to the original type.

Reply:
@MichaReiser Do you prefer reuse the name Arguments here? Then suggesting a new name for old Arg and the new Arg will be really helpful. Arguments can be named like `PythonArguments`

Reply:
We could be explicit about the fact that some nodes have defaults by naming the node `ArgWithDefault`.

Another alternative is to add `default` to `Arg`. This has the downside that it is now possible to construct invalid trees, but we might be fine with that. 

The part that irritates me about `FunctionArguments` is that the fact that the arguments are for functions is a given, Python only supports arguments for functions. 

My suggestion would be to go with `Arguments`. Maybe we can add some documentation to the README where we call out all differences to Python's AST? We could also add documentation to the Node but that's probably complicated because it is generated code.